packages/admin/admin/src/hooks/useAsyncOptionsProps.ts

```diff
@@ -15,17 +15,21 @@ export function useAsyncOptionsProps<T>(loadOptions: () => Promise<T[]>): AsyncO
     const [loading, setLoading] = useState(false);
     const [error, setError] = useState<Error | null>(null);
 
+    useEffect(() => {
+        setLoading(true);
+        loadOptions()
+            .then((newOptions) => {
+                setOptions(newOptions);
+            })
+            .catch((e) => {
+                setError(e);
+            })
+            .finally(() => {
+                setLoading(false);
+            });
+    }, [loadOptions]);
```

Comment:
This triggers a reload of the options every time the function changes. For instance, in the ProductForm this would trigger on every render of the form: https://github.com/vivid-planet/comet/blob/next/demo/admin/src/products/ProductForm.tsx#L213-L228. How would you address that?